extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/ssl/TypesafeGraphQLClientClientAuthenticationBadKeystoreTest.java

```diff
@@ -0,0 +1,89 @@
+package io.quarkus.smallrye.graphql.client.deployment.ssl;
+
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.graphql.Query;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.certs.Format;
+import io.smallrye.certs.junit5.Certificate;
+import io.smallrye.certs.junit5.Certificates;
+import io.smallrye.graphql.client.typesafe.api.GraphQLClientApi;
+import io.vertx.core.http.HttpServer;
+
+@Certificates(baseDir = "target/certs", certificates = {
+        @Certificate(name = "graphql", password = "password", formats = { Format.PKCS12 }, client = true),
+        @Certificate(name = "wrong-graphql", password = "wrong-password", formats = { Format.PKCS12 }, client = true)
+})
+public class TypesafeGraphQLClientClientAuthenticationBadKeystoreTest {
+
+    private static final int PORT = 63805;
+    private static final SSLTestingTools TOOLS = new SSLTestingTools();
+    private static HttpServer server;
+
+    private static final String CONFIGURATION = """
+                quarkus.smallrye-graphql-client.my-client.tls-configuration-name=my-tls-client
+                quarkus.tls.my-tls-client.key-store.p12.path=target/certs/wrong-graphql-client-keystore.p12
```

Comment:
Do you know what is supported by the client? P12, JKS seems to be supported. What about pem?

Reply:
As far as I know, SRGQL by itself supports only P12 and JKS since it works with path, password and type fields (see: https://github.com/smallrye/smallrye-graphql/blob/5ff2b59a0e16386ba03a51e47f908426667cd1c3/client/implementation/src/main/java/io/smallrye/graphql/client/impl/GraphQLClientConfiguration.java#L65-L78)

However, this new integration should support all three types 
Should I add separate tests for them (JKS/P12/PEM)?

But I talked with @jmartisk about possibly adding a feature to adding integration of PEM format in SRGQL (raw).

Reply:
It should support just whatever the TLS registry extension passes to us, AFAIU the GraphQL client shouldn't need to care about the underlying format, no? As long as we receive a `io.vertx.core.net.KeyCertOptions` or `TrustOptions`. I don't think we need to test each format separately on the GraphQL client side.

As for the original (very crude) support for keystores and truststores, I wouldn't care too much about updating it with extra formats, we clearly mark is as deprecated.